### ISSUE TITLE
Rendering broken

### ISSUE THREAD

**Issue by AxxlForce** [osxmidi/LinVst#42]
Hi,  

I freshly installed Ubuntu Studio 18.04. and I now got a rendering problem (see attachment). It won't show the VST's UI but instead artifacts of the underlying windows.

![screenshot_2018-07-12_23-15-36](https://user-images.githubusercontent.com/3379964/42659931-0a81e8fa-862a-11e8-88c7-67673ee91a5d.png)

Ubuntu Studio 18.04 x64
LinVst 2.2
wine-3.11 (Staging)

**Comment by lentferj** [comment]
Are you using linVST binaries or did you compile yourself?

**Comment by osxmidi** [comment]
I can't seem to replicate it myself, maybe it's a distro/wine/library problem.

**Comment by AxxlForce** [comment]
I'm using LinVst-2.2.2-Ubuntu17.10.zip from the releases. Yes I'm using 18.04 but this isn't the problem right?

**Comment by osxmidi** [comment]
Have you tried the Debian Stretch version 2.2.1, which is basically the same as the Ubuntu version, just made on Debian.

I'll take down the Ubuntu version for the time being.

**Comment by fast4shoot** [comment]
I am experiencing the same problem on Arch Linux with LinVst built from source using [this AUR package](https://aur.archlinux.org/packages/linvst/). The standalone version works fine.

I tested it in both Reaper and Carla.

As you say, it's probably a distro/wine/library problem, so that makes it hard to pin down.

**Comment by osxmidi** [comment]
I've put up a version with the video delay for Debian/Ubuntu on the Releases page if anyone wants to test it out and let me know.

LinVst-2.2.1-Debian-Stretch-Delay.zip

**Comment by fast4shoot** [comment]
I tried that out and it has sort of worked in Carla. The only problem is that, for some reason, the GUI is floating above everything without a title bar and in some cases (PG-8X) cannot be interacted with.

The situation is still the same in Reaper.

**Comment by fast4shoot** [comment]
So it appears that I've been running wine-staging. I installed regular wine and the GUIs started working fine in Reaper. So I installed wine-staging again and they keep working, even without EMBEDTHREAD. I'm not sure what version I had installed before I did all this, so I don't know if it got fixed by updating to a newer version or something else.

Carla, OTOH, is still the same - nothing shows up without EMBEDTHREAD defined and a floating GUI without a title bar shows up with EMBEDTHREAD defined.

FWIW, I'm running kernel 4.17.5, xorg-server 1.20.0, xf86-video-amdgpu 18.0.1 with R9 290 and wine-staging 3.12

**Comment by fast4shoot** [comment]
>  it's a wrapper inside another wrapper 

What do you mean? I thought Carla's support for VSTs was native, is that not the case?

**Comment by osxmidi** [comment]
What happens is that the Linux vst reparents it's window to the window supplied by the host.

As far as the windows dll loaded by LinVst is concerned, LinVst is the host.

If linvst.so (or and Linux vst .so file) is opened by Carla then Carla is the host as far as LinVst is concerned and Carla is supplying a window that LinVst reparents a window into (which is a wine window) and at the same time Carla is between the DAW and LinVst as Carla was loaded by the DAW host so LinVst (and the windows vst) needs to go through Carla to get to the DAW host and the DAW host needs to go through Carla to get to LinVst (and the windows vst).

Basically it just complicates things and things like GUI artifacts might happen.

**Comment by fast4shoot** [comment]
I see what you mean, though I've been using Carla standalone.

Anyway, it seems that I have spoken too son about LinVst in Reaper working, because it seems like it has stopped working again, with or without EMBEDTHREAD. Not sure what happened - I haven't done much to the system, just updated a couple of packages.
